From a3cfdde7f2f601ae6efdab125dc0bee0689ae224 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:10:55 -0400
Subject: [powerpc] Fix ABIv2 issue with dereference_function_descriptor

Message-id: <1410545655-205645-427-git-send-email-dzickus@redhat.com>
Patchwork-id: 94609
O-Subject: [RHEL7 PATCH 426/626] powerpc: Fix ABIv2 issue with dereference_function_descriptor
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 07de8377f7488f262f9694a1567ab93b4dda63bc
Author: Anton Blanchard <anton@samba.org>
Date:   Tue Mar 11 12:15:27 2014 +1100

    powerpc: Fix ABIv2 issue with dereference_function_descriptor

    Don't try and dereference a function descriptor on ABIv2.

    Signed-off-by: Anton Blanchard <anton@samba.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/include/asm/sections.h b/arch/powerpc/include/asm/sections.h
index d0e784e..d1bb96d 100644
--- a/arch/powerpc/include/asm/sections.h
+++ b/arch/powerpc/include/asm/sections.h
@@ -39,6 +39,7 @@ static inline int overlaps_kernel_text(unsigned long start, unsigned long end)
   (unsigned long)_stext < end;
 }
 
+#if !defined(_CALL_ELF) || _CALL_ELF != 2
 #undef dereference_function_descriptor
 static inline void *dereference_function_descriptor(void *ptr)
 {
@@ -49,6 +50,7 @@ static inline void *dereference_function_descriptor(void *ptr)
   ptr = p;
  return ptr;
 }
+#endif
 
 #endif
 
-- 
1.7.1